lib/gakubuchi/engine_registrar.rb

```diff
@@ -0,0 +1,26 @@
+module Gakubuchi
+  class EngineRegistrar
+    def initialize(env)
+      @env = env
+    end
+
+    def register!(target, engine)
+      klass = constantize(engine)
+      return false if !klass || registered?(target)
+
+      @env.register_engine(target, klass)
+      true
+    end
+
+    def registered?(target)
+      @env.engines.key?(::Sprockets::Utils.normalize_extension(target))
+    end
+
+    private
+
+    def constantize(klass)
+      return klass if klass.instance_of?(::Class)
```

Comment:
[IMO]
This line seems redundant because `klass.to_s.constantize` returns an instance of `Class` when `klass` is an instance of `Class`.